runtime/src/test/java/com/tietoevry/quarkus/resteasy/problem/javax/ConstraintViolationExceptionMapperTest.java

```diff
@@ -20,63 +28,169 @@
+                .usingFieldByFieldElementComparator()
+                .contains(
+                        Violation.In.BODY.violation("length must be between 6 and 10", "param_name_from_annotation"),
+                        Violation.In.BODY.violation("length must be between 7 and 10", "items[1].param_name_from_annotation"),
+                        Violation.In.BODY.violation("length must be between 8 and 10", "inner.param_name_from_annotation"));
+    }
+
+    private List<Violation> mapAndExtractViolations(ConstraintViolationException exception) {
         Response response = mapper.toResponse(exception);
 
         assertThat(response.getStatus()).isEqualTo(400);
         assertThat(response.getMediaType()).isEqualTo(HttpProblem.MEDIA_TYPE);
         HttpProblem problem = (HttpProblem) response.getEntity();
-        List<Violation> violations = (List<Violation>) problem.getParameters().get("violations");
-        assertThat(violations)
-                .usingFieldByFieldElementComparator()
-                .containsExactlyInAnyOrder(
-                        Violation.inQuery("length must be between 10 and 15", "param_name"),
-                        Violation.inHeader("length must be between 10 and 15", "param_name3"),
-                        Violation.inPath("length must be between 10 and 15", "param_name4"),
-                        Violation.inBody("must be greater than or equal to 15", "phraseName"),
-                        Violation.inBody("must be greater than or equal to 15", "innerObj.phraseName"));
+        return (List<Violation>) problem.getParameters().get("violations");
     }
 
-    private ConstraintViolationException createValidationException() {
-        final Set<ConstraintViolation<Object>> constraintViolations = VALIDATOR.forExecutables()
-                .validateParameters(
-                        resourceInfoStub,
-                        resourceInfoStub.getResourceMethod(),
-                        new Object[] { "a", "a", "a", "a", new ResourceInfoStub.StubBody() },
-                        new Class[0]);
+    static class StubResourceInfo implements ResourceInfo {
 
-        return new ConstraintViolationException(constraintViolations);
-    }
+        private final Validator validator;
 
-    static class ResourceInfoStub implements ResourceInfo {
+        StubResourceInfo(Validator validator) {
+            this.validator = validator;
+        }
+
+        public static StubResourceInfo withDefaultValidator() {
+            return new StubResourceInfo(Validation.byProvider(HibernateValidator.class)
+                    .configure()
+                    .buildValidatorFactory()
+                    .getValidator());
+        }
+
+        public static StubResourceInfo withJsonPropertyAwareValidator() {
+            return new StubResourceInfo(Validation.byProvider(HibernateValidator.class)
+                    .configure()
+                    .parameterNameProvider(new ParanamerParameterNameProvider())
```

Comment:
Here @pazkooda ;) Hibernate already provides part of Paranamer as transitive dependency, but without explicitly adding it to pom.pl I get runtime exceptions (class not found)

Reply:
Well, with package name starting with `org.hibernate.validator` and `groupId` being `com.thoughtworks.paranamer` might be difficult to spot in PullReq view.
Also it a bit strange that they "borrow" Java `package` from other library - can cause theoretical collisions in future 😕 .

Reply:
Hibernate can use paranamer library via `ParanamerParameterNameProvider` - it's an adapter from paranamer world to the Hibernate one :)